test/taskrun_test.go

```diff
@@ -0,0 +1,146 @@
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	hwTaskName    = "helloworld"
+	hwTaskRunName = "helloworld-run"
+)
+
+func getHelloWorldTask(namespace string) *v1alpha1.Task {
+	return &v1alpha1.Task{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      hwTaskName,
+		},
+		Spec: v1alpha1.TaskSpec{
+			BuildSpec: &buildv1alpha1.BuildSpec{
+				Steps: []corev1.Container{
+					corev1.Container{
+						Name:  "helloworld-busybox",
+						Image: "busybox",
+						Args: []string{
+							"echo", "hello world",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getHelloWorldTaskRun(namespace string) *v1alpha1.TaskRun {
+	return &v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      hwTaskRunName,
+		},
+		Spec: &v1alpha1.TaskRunSpec{
+			TaskRef: v1alpha1.TaskRef{
+				Name: hwTaskName,
+			},
+			Trigger: v1alpha1.TaskTrigger{
+				TriggerRef: v1alpha1.TaskTriggerRef{
+					Type: v1alpha1.TaskTriggerTypeManual,
+				},
+			},
+		},
+	}
+}
+
+// TestTaskRun is an integration test that will verify a very simple "hello world" TaskRun can be
+// executed.
+func TestTaskRun(t *testing.T) {
+	t.Skip("Will fail until #59 is completed :D")
+	// TODO: when committing, actually skip this test since it won't pass yet
+
+	logger := logging.GetContextLogger(t.Name())
+	c, namespace := setup(t, logger)
+
+	knativetest.CleanupOnInterrupt(func() { tearDown(logger, c.KubeClient, namespace) }, logger)
+	defer tearDown(logger, c.KubeClient, namespace)
+
+	// Create task
+	_, err := c.TaskClient.Create(getHelloWorldTask(namespace))
+	if err != nil {
+		t.Fatalf("Failed to create Task `%s`: %s", hwTaskName, err)
+	}
+
+	// Create TaskRun
+	_, err = c.TaskRunClient.Create(getHelloWorldTaskRun(namespace))
+	if err != nil {
+		t.Fatalf("Failed to create TaskRun `%s`: %s", hwTaskRunName, err)
+	}
+
+	// Verify status of TaskRun (wait for it)
+	err = WaitForTaskRunState(c, hwTaskRunName, func(tr *v1alpha1.TaskRun) (bool, error) {
+		if tr.Status != nil && len(tr.Status.Conditions) > 0 {
+			// TODO: use actual conditions
+			return true, nil
+		}
+		return false, nil
+	}, "TaskRunCompleted")
+	if err != nil {
+		t.Errorf("Error waiting for TaskRun %s to finish: %s", hwTaskRunName, err)
+	}
+
+	// Verify logs
+	// make sure it output hello world
+	// get pod from taskrun
+	// taskrun -> build
+	b, err := c.BuildClient.Get(hwTaskRunName, metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("Expected there to be a Build with the same name as TaskRun %s but got error: %s", hwTaskRunName, err)
+	}
+	// build -> pod
+	cluster := b.Status.Cluster
+	if cluster == nil || cluster.PodName == "" {
+		t.Fatalf("Expected build status to have a podname but it didn't!")
+	}
+	// pod -> logs
+	podName := cluster.PodName
+
+	pods := c.KubeClient.Kube.CoreV1().Pods(namespace)
+
+	fmt.Printf("I GOT PODS %s %s\n", podName, pods)
```

Comment:
is there a mapping between build to pod? we can look for pods starting with expected name and error out if they don't

Reply:
yep! afaik in @aaron-prindle 's implementation, the names of the `Builds` are the same as the names of the `TaskRuns`. here we are using that to get the `Build`: `b, err := c.BuildClient.Get(hwTaskRunName...` (I'll add a comment about this), and then the `Cluster` is an attribute on `Build` which we can use to find the `Pod` - the next challenge is finding the right containers :weary:

Reply:
The build pods are terminated when the build is done

Reply:
:(

Reply:
hmm but even now it's possible to get logs from the pods the Build ran on, even if the Build has completed right? (is it the case that it's a race?)

Reply:
You can’t get build logs after the build is done unless you have been saving them somewhere else